js/repeater.js

```diff
@@ -617,48 +666,34 @@
 				this.$loader.show().loader('play');
 			}
 
-			dataOptions = this.getDataOptions(options);
-
-			this.viewOptions.dataSource(dataOptions, function (data) {
-				data = data || {};
+			var dataOptions = this.getDataOptions(options);
 
-				if (self.infiniteScrollingEnabled) {
-					// pass empty object because data handled in infiniteScrollPaging method
-					self.infiniteScrollingCallback({});
-				} else {
-					self.itemization(data);
-					self.pagination(data);
-				}
-
-				self.runRenderer(viewTypeObj, data, function () {
-					if (self.infiniteScrollingEnabled) {
-						if (viewChanged || options.clearInfinite) {
-							self.initInfiniteScrolling();
+			var beforeRender = this.viewOptions.dataSource;
+			var self = this;
```

Comment:
I know this is standard, but something like `repeaterPrototypeContext` would be most clear.